src/containers/Signup/Signup.tsx

```diff
@@ -24,65 +41,72 @@ const INITIAL_FORM_FIELDS: IFormFields = {
   confirmationCode: ''
 };
 
-const Signup: FC<IAppProps & RouteComponentProps> = props => {
+const Signup: FC<RouteComponentProps> = ({ history }) => {
+  const { push: historyPush } = history;
+  const dispatch = useDispatch();
+
+  const signingUp = useSelector(selectAuthSigningUp);
+  const signedUp = useSelector(selectAuthSignedUp);
+  const signUpError = useSelector(selectAuthSignUpError);
+
+  const activating = useSelector(selectAuthActivating);
+  const activated = useSelector(selectAuthActivated);
+  const activateError = useSelector(selectAuthActivateError);
+
+  const signingIn = useSelector(selectAuthSigningIn);
+  const signedIn = useSelector(selectAuthSignedIn);
+  const signInError = useSelector(selectAuthSignInError);
+
   const [fields, handleFieldChange] = useFormFields<IFormFields>(
     INITIAL_FORM_FIELDS
   );
-  const [newUser, setNewUser] = useState<ISignUpResult | null>(null);
-  const [isLoading, setIsLoading] = useState(false);
+
+  const { email, password } = fields;
+
+  useEffect(() => {
+    if (signUpError) alert(signUpError);
+    if (activateError) alert(activateError);
+    if (signInError) alert(signInError);
+  }, [signUpError, activateError, signInError]);
```

Comment:
This looks like actually just 
```
type MyError = SignUpError | ActivateError | SignError;
errors.map((error: MyError) => alert(error))
```

Reply:
It's actually happening one of, not all of it is truthy, only one of those is truthy in a render. If i did map each error will fire alert. But i didn't know you could map type lol

Reply:
I don't really understand what you said. I mean that your selector could be something like getErrors, and since you do not apply special logic, but simply alert if any error exist, without over complicating things.